src/shared/telemetry/defaultTelemetryService.ts

```diff
@@ -242,16 +244,77 @@ export class DefaultTelemetryService implements TelemetryService {
         return event
     }
 
-    private static readEventsFromCache(cachePath: string): TelemetryEvent[] {
+    public static readEventsFromCache(cachePath: string): TelemetryEvent[] {
         try {
-            const events = JSON.parse(fs.readFileSync(cachePath, 'utf-8')) as TelemetryEvent[]
+            const input = JSON.parse(fs.readFileSync(cachePath, 'utf-8'))
+            const events = this.filterTelemetryCacheEvents(input)
             events.forEach((element: TelemetryEvent) => {
                 element.createTime = new Date(element.createTime)
```

Comment:
Side question: Do we want this to track the time the telemetry data was popped out of the cache or the time that the telemetry entered the cache? I feel like this would be misleading.